src/silx/gui/plot/actions/histogram.py

```diff
@@ -398,13 +405,19 @@ def _updateFromItem(self):
             array.ravel().astype(numpy.float32),
             n_bins=max(2, self.__nbinsLineEdit.getValue()),
             histo_range=self.__rangeSlider.getValues(),
+            weights=array.ravel().astype(numpy.float32),
```

Comment:
maybe we could create an intermediary array like
```
data = array.ravel().astype(numpy.float32)
        histogram = Histogramnd(
            data,
            n_bins=max(2, self.__nbinsLineEdit.getValue()),
            histo_range=self.__rangeSlider.getValues(),
            weights=data,
        )
``` 
as we are using it twice this would avoid to do the operation several time.

Reply:
OK